exercises/practice/high-scores/.articles/oop/content.md

```diff
@@ -0,0 +1,142 @@
+# Introduction
+
+Object oriented programming in Lua uses the same concepts as OOP in other languages: classes, instances, methods, etc.
+The implementation of a class is done with plain Lua tables and functions, and (perhaps a new concept for you) _metatables_.
+
+A quick example: let's create a `Car` class that can create new cars for a specified manufacturer
+
+```lua
+local Car = {}
+Car.__index = Car
+
+function Car.new(class, manufacturer)
+    local new_car = {}
+    new_car.manufacturer = manufacturer
+    setmetatable(new_car, class)
+    return new_car
+end
+
+my_car = Car.new(Car, "Volvo")
+print(my_car.manufacturer)   -- Volvo
+```
+
+We can see the class is just a table with one member, a function named `new`.
+The `new` function creates a table, adds a member to it, does some magic, and then returns the table.
+We will see what `__index` and `setmetatable` do a little further down.
+This "new" function is acting as the constructor for the class, creating a new instance.
+
+~~~~exercism/note
+Why did I have to pass `Car` as the first argument to `new`?
+This is a convention to enable something called "colon notation".
+
+## Colon notation
+
+Lua has a bit of "syntactic sugar" that allows us to have a more OO-like syntax.
+We can write
+
+```lua
+my_car = Car.new(Car, "Volvo")
+```
+
+or, with colon notation, we can write
+
+```lua
+my_car = Car:new("Volvo")
+```
+
+The colon notation can pass the left-hand side of the colon as the first argument.
+
+The constructor itself can be written with colon notation.
+In this case `self` is enforced to be the name of the class.
+
+```lua
+function Car:new(manufacturer)
+    local new_car = {}
+    new_car.manufacturer = manufacturer
+    setmetatable(new_car, self)         -- <<< here
+    return new_car
+end
+```
+
+We will be defining methods using this colon notation.
+~~~~
+
+## Methods
+
+We have a new car, but it can't do anything.
+Let's add a method to the class.
+
+```lua
+function Car:honk()
+    print("The " .. self.manufacturer .. " car goes: beep!")
+end
+
+my_car:honk()       -- The Volvo car goes: beep!
+```
+
+How does the car instance know where to find the `honk` method?
+The Car "new" constructor did not add a honk function to the instance table.
+
+### Method lookup
+
+This is where **metatables** come in to the picture.
+
+It goes like this.
+
+1. Does `my_car` have a `honk` member? No.
+1. Does `my_car` have a metatable? Yes, we set the metatable for `my_car` to be `Car`.
+1. Does the metatable have an `__index` element? Yes, we wrote `Car.__index = Car`
+1. Does `Car` have a `honk` member? Yes!
```

Comment:
```suggestion
1. Does `Car` (the `__index` element) have a `honk` member? Yes!
```

I think this might make it slightly clearer why we're looking in `Car`